src/components/date.js

```diff
@@ -4,44 +4,50 @@
  * See the accompanying LICENSE file for terms.
  */
 
-import React, {Component} from 'react';
-import PropTypes from 'prop-types';
-import withIntl from './withIntl';
-import {intlShape, dateTimeFormatPropTypes} from '../types';
-import {invariantIntlContext, shouldIntlComponentUpdate} from '../utils';
-
-class FormattedDate extends Component {
-  static displayName = 'FormattedDate';
-
-  static propTypes = {
-    ...dateTimeFormatPropTypes,
-    intl: intlShape,
-    value: PropTypes.any.isRequired,
-    format: PropTypes.string,
-    children: PropTypes.func,
-  };
-
-  constructor(props) {
-    super(props);
-    invariantIntlContext(props);
-  }
-
-  shouldComponentUpdate(nextProps, nextState) {
-    return shouldIntlComponentUpdate(this, nextProps, nextState);
-  }
-
-  render() {
-    const {formatDate, textComponent: Text} = this.props.intl;
-    const {value, children} = this.props;
-
-    let formattedDate = formatDate(value, this.props);
+import React, { useMemo } from 'react'
+import PropTypes from 'prop-types'
+import useIntl from './useIntl'
+import { dateTimeFormatPropTypes } from '../types'
+
+function FormattedDate(props) {
+  const { formatDate, textComponent: Text } = useIntl()
+
+  const { value, children } = props
+
+  return useMemo(() => {
+    let formattedDate = formatDate(value, props)
 
     if (typeof children === 'function') {
-      return children(formattedDate);
+      return children(formattedDate)
```

Comment:
sorta wonder if these are still relevant... like, wouldn't a `useFormattedDate` hook give us equivalent functionality? i guess it doesn't hurt